feat: add a @userschema/registration endpoint (#1874)
plone/plone.restapi: docs/source/endpoints/userschema.md

The user schema uses the same serialization as the type's JSON schema.

See {ref}`types-schema` for detailed documentation about the available field types.

> ## Getting the registration form

Reviewer: ```suggestion
## Get the registration form
```

> In Plone we can configure each of the fields of the user schema to be available only in the user edit form, in the registration form or in both of them.

Reviewer: I think this is your intent, but please verify.

```suggestion
In Plone you can configure each of the fields of the user schema to be available in only one of either the user edit form or registration form, or in both of them.
```

Author: that's right.

> To get the user schema available for the user registration form, make a request to the `/@registration-userschema` endpoint.

Reviewer: Per https://github.com/plone/plone.restapi/issues/1873#issuecomment-2625169067.

```suggestion
To get the user schema available for the user registration form, make a request to the `@userschema/registration` endpoint.
```

```{eval-rst}
..  http:example:: curl httpie python-requests

>     :request: ../../../src/plone/restapi/tests/http-examples/registration_userschema.req

Reviewer: Depends on refactoring of filepath.

```suggestion
    :request: ../../../src/plone/restapi/tests/http-examples/userschema/registration.req
```

```

The server will respond with the user schema.

> ```{literalinclude} ../../../src/plone/restapi/tests/http-examples/registration_userschema.resp

Reviewer: ```suggestion
```{literalinclude} ../../../src/plone/restapi/tests/http-examples/userschema/registration.resp
```
